### PR TITLE
DALA-302 modified the set request header to include minimum proofing …

### PR DESCRIPTION
**Pull request by interestinginteger123**
…level

## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

**Comment by Sourceless** [comment]
ExtractVariable is failing when it can't find the custom attribute it's expecting -- should there be something on the test app? Isn't this supposed to work even if the custom attribute is not present?

File: proxies/live/apiproxy/policies/ExtractVariables.GetProofingLevel.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<ExtractVariables name="ExtractVariables.GetProofingLevel" continueOnError="true">
+  <!-- The source element points to the variable populated by AccessEntity policy.
+  The format is <policy-type>.<policy-name>.
+  In this case, the variable contains the whole developer profile. -->
+  <Source>AccessEntity.AccessEntity.GetApp</Source>
+  <VariablePrefix>developer.app</VariablePrefix>
+  <XMLPayload>
+    <Variable name="nhs-login-allowed-proofing-level" type="string">
+      <!-- You parse elements from the developer profile using XPath. -->
+      <XPath>/App/Attributes/Attribute[Name='nhs-login-allowed-proofing-level']/Value/text()</XPath>
```
Comment (Marcin-Duszynski) [inline proxies/live/apiproxy/policies/ExtractVariables.GetProofingLevel.xml:10]: Should this parameter name nhs-login-minimum-proofing-level?

Comment (vurt007) [reply proxies/live/apiproxy/policies/ExtractVariables.GetProofingLevel.xml:10]: no, spec says the 'custom attribute is :
"nhs-login-allowed-proofing-level"
the variable passed to the backend is:
"developer.app.nhs-login-minimum-proofing-level"

https://nhsd-confluence.digital.nhs.uk/display/DSA/1.+Patient+Vaccine+History#id-1.PatientVaccineHistory-APIMFunctionality

File: proxies/live/apiproxy/resources/jsc/ClientRPDetailsHeader.SetRequestHeaders.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
 var developerAppName = context.getVariable('developer.app.name');
 var developerAppId = context.getVariable('developer.app.id');
 var clientIP = context.getVariable('client.ip');
+var allowedProofingLevel = context.getVariable('developer.app.nhs-login-allowed-proofing-level');
+
+if (allowedProofingLevel === null) {
+    allowedProofingLevel = "P9";
```
Comment (vurt007) [inline proxies/live/apiproxy/resources/jsc/ClientRPDetailsHeader.SetRequestHeaders.js:7]: this shouldn't default to P9 ...  the backend should deal with that .. possibly it should default to null / empty string
(there are future use-cases that will be outside the P5/P9 options)

File: proxies/live/apiproxy/policies/ExtractVariables.GetProofingLevel.xml (hunk 1 above)
Comment (Marcin-Duszynski) [reply proxies/live/apiproxy/policies/ExtractVariables.GetProofingLevel.xml:10]: thx @vurt007

**Review by Sourceless — APPROVED** [review]

**Pull request merged**